internal/graph/graph_test.go

```diff
@@ -451,6 +451,30 @@ func TestShortenFunctionName(t *testing.T) {
 			"github.com/blah/blah/vendor/gopkg.in/redis.v3.(*baseClient).(github.com/blah/blah/vendor/gopkg.in/redis.v3.process)-fm",
 			"redis.v3.(*baseClient).(github.com/blah/blah/vendor/gopkg.in/redis.v3.process)-fm",
 		},
+		{
+			"github.com/foo/bar/v4.(*Foo).Bar",
```

Comment:
Curious - why is version string sometimes a separate subdirectory and sometimes a prefix of the package name? Is this something that the package owners choose? Are these options restricted at these two, or are there more?

Oh, I guess it's a function of how deep below the versioning level the actual symbol is?

Reply:
I think you're referring to the tests; some of the tests I've added are where the "version" isn't a version at all. The only valid version in paths are "v2", "v3", ... "v1234", etc. So you'd have `github.com/foo/bar`, `github.com/foo/bar/v2`, `github.com/foo/bar/v3`, and so on, then subpackages like `github.com/foo/bar/v3/baz`. Custom domains mean you can have things like `gotest.tools/assert`, `gotest.tools/v3/assert`. But a package can be at the level where the version appears, so when `github.com/jackc/pgx` was bumped to `github.com/jackc/pgx/v4`, it's still referred to as `pgx` in the code.

But if it isn't a valid version part, then I don't want to treat it as one naively (i.e. "something.com/hello/v123xyz" isn't versioned, "something.com/hello/v123/xyz" is because the version is its own element).